### PR TITLE
]MODEL-12503] insert secrets values into fit/score

### PR DESCRIPTION
**Pull request by eric-s-s**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

This PR adds code to handle injecting secrets for custom tasks.  We want to make sure that the user can use the secrets but that they cannot get them from the artifact.  This will prevent other users from having access to the author's secrets.

This adds changes and tests all through the call stack to successfully implement this.

## Incidental changes

- updated an enum TargetType with more clear usage
- moved all model adapters to same directory under adapters
- made an ABC for model adapters so that the interface where `fit` is called is always correct
- removed the optional args for fit

### CODE REVIEW

File: custom_model_runner/datarobot_drum/resource/drum_server_utils.py [hunk 1]
```diff
@@ -87,65 +91,35 @@ def __init__(
         else:
             self.url_server_address = "http://localhost:{}".format(self.port)
 
-        log_level = logging.getLevelName(logging.root.level).lower()
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/resource/drum_server_utils.py:90]: code moved as-is into separate method for easier testing

File: testlist.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
+- tests/unit/datarobot_drum/drum/test_drum.py: OK
```
Comment (eric-s-s) [inline testlist.txt:1]: TODO remove this!!!

**Comment by eric-s-s** [comment]
jarvis please test this

**Review by scottp-ml — APPROVED** [review]

**Comment by eric-s-s** [comment]
jarvis please test test_training_model_templates

**Comment by eric-s-s** [comment]
jarvis please test test_training_model_templates

**Comment by eric-s-s** [comment]
jarvis test test_integration_java_environment

**Comment by eric-s-s** [comment]
jarvis please test this

**Comment by eric-s-s** [comment]
jarvis test_unit

**Comment by eric-s-s** [comment]
jarvis test test_unit

**Comment by eric-s-s** [comment]
jarvis please test this

**Comment by eric-s-s** [comment]
jarvis test test_integration_general

**Comment by eric-s-s** [comment]
jarvis test test_integration_r_environment

**Comment by eric-s-s** [comment]
jarvis test test_drop_in_envs

**Comment by eric-s-s** [comment]
jenkins run continuous-integration/jenkins/pr-merge

**Pull request merged**